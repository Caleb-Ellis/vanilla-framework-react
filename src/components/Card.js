import React from 'react';

class Card extends React.Component {

  render() {
    return (
      <div className={ this.props.modifier ? 'p-card--' +  this.props.modifier : 'p-card' }>
        <header className="p-card__header">
          <img src={ this.props.imgSrc } alt="placeholder" />
        </header>
        <h3 className="p-card__title">{ this.props.title }</h3>
        <p className="p-card__content">{ this.props.cardContent }</p>
      </div>
    )
  }
}

export default Card;
