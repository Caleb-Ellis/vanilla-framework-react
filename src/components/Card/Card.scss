@import 'vanilla-framework/scss/patterns_card';
@include vf-p-card;