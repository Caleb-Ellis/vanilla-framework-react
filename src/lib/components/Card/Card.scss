@import 'vanilla-framework/scss/base';
@import 'vanilla-framework/scss/patterns_card';
@include vf-b-typography;
@include vf-p-card;