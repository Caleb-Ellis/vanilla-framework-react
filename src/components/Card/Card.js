import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';

class Card extends React.Component {
  render() {
    return (
      <div className={this.props.modifier ? `p-card--${this.props.modifier}` : 'p-card'}>
        <header className="p-card__header">
          <img src={this.props.imgSrc} alt={this.props.imgAlt} />
        </header>
        <h3 className="p-card__title">{this.props.title}</h3>
        <p className="p-card__content">{this.props.cardContent}</p>
      </div>
    );
  }
}

Card.defaultProps = {
  modifier: '',
  imgSrc: 'http://placehold.it/200x32',
  imgAlt: 'Placeholder',
};

Card.propTypes = {
  modifier: PropTypes.string,
  imgSrc: PropTypes.string,
  imgAlt: PropTypes.string,
  title: PropTypes.string.isRequired,
  cardContent: PropTypes.string.isRequired,
};

export default Card;
