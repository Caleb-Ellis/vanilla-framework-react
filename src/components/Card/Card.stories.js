import React from 'react';
import { storiesOf } from '@storybook/react';

import Card from './Card';

storiesOf('Card', module)
  .add('Default', () => (
    <Card
      title="Card title"
      image={{
        src: 'http://placekitten.com/g/64/64',
        alt: 'Placeholder',
      }}
    >
      <p>Lorem ipsum dolor sit amet, consectetur adipisicing.</p>
    </Card>
  ))
  .add('Highlighted', () => (
    <Card
      modifier="highlighted"
      title="Card title"
      image={{
        src: 'http://placekitten.com/g/64/64',
        alt: 'Placeholder',
      }}
    >
      <p>Lorem ipsum dolor sit amet, consectetur adipisicing.</p>
    </Card>
  ));
