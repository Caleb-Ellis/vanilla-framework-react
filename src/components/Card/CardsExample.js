import React from 'react';
import Card from './Card';

class CardsExample extends React.Component {
  constructor() {
    super();
    this.image = {
      src: 'http://placekitten.com/g/64/64',
      alt: 'Placeholder',
    };
  }
  render() {
    return (
      <div>
        <h2>Card</h2>
        <Card
          title="Card title"
          image={this.image}
        >
          <p>Lorem ipsum dolor sit amet, consectetur adipisicing.</p>
        </Card>
        <h2>Highlighted Card</h2>
        <Card
          modifier="highlighted"
          title="Card title"
          image={this.image}
        >
          <p>Lorem ipsum dolor sit amet, consectetur adipisicing.</p>
        </Card>
      </div>
    );
  }
}

export default CardsExample;
