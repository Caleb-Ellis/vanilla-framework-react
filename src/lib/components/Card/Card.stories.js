import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import Card from './Card';
import Column from '../Strip/Column';
import Row from '../Strip/Row';
import Strip from '../Strip/Strip';

storiesOf('Card', module)
  .add('Default',
    withInfo('The purpose of the basic card is to display information, without user interaction.')(() => (
      <Card title="Card title">
        <p>Lorem ipsum dolor sit amet, consectetur adipisicing.</p>
      </Card>),
    ),
  )

  .add('With Header',
    withInfo('Card components accept either a header or image prop for an optional header.')(() => (
      <Card header="Card header" title="Card title">
        <p>Lorem ipsum dolor sit amet, consectetur adipisicing.</p>
      </Card>),
    ),
  )

  .add('With Image',
    withInfo('Card components accept either a header or image prop for an optional header.')(() => (
      <Card
        title="Card title"
        image={{
          src: 'http://placekitten.com/g/64/64',
          alt: 'Placeholder',
        }}
      >
        <p>Lorem ipsum dolor sit amet, consectetur adipisicing.</p>
      </Card>),
    ),
  )

  .add('Highlighted',
    withInfo('The highlighted card should be used when you can interact with the content.')(() => (
      <Card
        highlighted
        title="Card title"
        image={{
          src: 'http://placekitten.com/g/64/64',
          alt: 'Placeholder',
        }}
      >
        <p>Lorem ipsum dolor sit amet, consectetur adipisicing.</p>
      </Card>),
    ),
  )

  .add('Overlay',
    withInfo('The purpose of the overlay prop is to make the text visible in conjunction with a Strip image component.')(() => (
      <Strip image={{
        src: 'https://assets.ubuntu.com/v1/ab1a7e82-background.png?w=600',
        colour: 'dark',
        }}
      >
        <Row>
          <Column size={6}><div /></Column>
          <Column size={6}>
            <Card
              overlay
              title="Card title"
              image={{
                src: 'http://placekitten.com/g/64/64',
                alt: 'Placeholder',
              }}
            >
              <p>Lorem ipsum dolor sit amet, consectetur adipisicing.</p>
            </Card>
          </Column>
        </Row>
      </Strip>),
    ),
  );
