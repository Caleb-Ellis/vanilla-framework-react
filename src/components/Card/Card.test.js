import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Card from './Card';

describe('Card component should render expected markup', () => {
  it('should compare the component with a snapshot', () => {
    const card = ReactTestRenderer.create(<Card
      className="p-card"
      title="Title"
      cardContent="Lorem ipsum dolor sit amet"
    />);
    const json = card.toJSON();
    expect(json).toMatchSnapshot();
  });
});
