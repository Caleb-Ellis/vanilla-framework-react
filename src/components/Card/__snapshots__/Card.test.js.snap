// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render expected markup should compare the component with a snapshot 1`] = `
<div
  className="p-card"
>
  <header
    className="p-card__header"
  >
    <img
      alt="Placeholder"
      src="http://placehold.it/200x32"
    />
  </header>
  <h3
    className="p-card__title"
  >
    Title
  </h3>
  <p
    className="p-card__content"
  >
    Lorem ipsum dolor sit amet
  </p>
</div>
`;
